### PR TITLE
Added new dialog system variables + more (tutorial) dialog

### PR DESCRIPTION
**Pull request by dansebmey**
Added new variables to the dialog system:
- DialogueGroup.initial_delay: postpones conversation until X seconds have passed
- DialogueSequence.max_repetitions: prevents sequences from being repeated more than X times
- DialogueSingleEntry.fixed_duration: automatically continues the dialog after X seconds (temporary fix for slow dialog)

Also added tutorial dialog entries, giving info about (central) rosettes, capturing pieces, and reaching the finish.

Last but not least, some more general dialog for flavour!

### CODE REVIEW

File: scenes/game/dice/d4_die.tscn [hunk 1]
```diff
@@ -1,11 +1,9 @@
-[gd_scene load_steps=17 format=3 uid="uid://bfyhvb7v43pnc"]
```
Comment (Marc313) [inline scenes/game/dice/d4_die.tscn:1]: Revert this whole scene

File: scenes/game/singleplayer.tscn [hunk 2]
```diff
@@ -173,20 +173,22 @@ script = ExtResource("6_3dit1")
 [node name="Opponent_Guard_Animated" parent="." instance=ExtResource("7_tegfk")]
 transform = Transform3D(0.848048, 0, 0.52992, 0, 1, 0, -0.52992, 0, 0.848048, 4.17257, -0.684179, 5.80273)
 
-[connection signal="game_ended" from="Boardgame/GameStateController" to="Prototype_UI" method="_on_game_ended"]
```
Comment (Marc313) [inline scenes/game/singleplayer.tscn:176]: Not sure if these signals should be removed

File: scenes/game/singleplayer.tscn [hunk 3]
```diff
@@ -173,20 +173,22 @@ script = ExtResource("6_3dit1")
...
 [connection signal="pressed" from="Prototype_UI/Main Menu/Play" to="Prototype_UI" method="_on_game_started"]
 [connection signal="pressed" from="Prototype_UI/Main Menu/Quit" to="Prototype_UI/Main Menu/Quit" method="_on_pressed"]
 [connection signal="on_opponent_ready" from="Opponent_Guard_Animated" to="Custom_Camera" method="_on_play_pressed"]
+[connection signal="on_prevent_opponent_action" from="Opponent_Guard_Animated/DialogueSystem/DialogueSequencePlayer" to="Boardgame/AIRoller" method="_disable_rolling"]
+[connection signal="on_resume_opponent_action" from="Opponent_Guard_Animated/DialogueSystem/DialogueSequencePlayer" to="Boardgame/AIRoller" method="_enable_rolling"]
 
 [editable path="Boardgame"]
 [editable path="Boardgame/Board"]
 [editable path="Boardgame/Dice"]
+[editable path="Opponent_Guard_Animated"]
```
Comment (Marc313) [inline scenes/game/singleplayer.tscn:194]: Does it need to be editable?

File: scripts/ai/ai_player_advanced.gd [hunk 4]
```diff
@@ -63,6 +63,7 @@ func _calculate_base_score(move: Move):
 	else:
```
Comment (Marc313) [inline scripts/ai/ai_player_advanced.gd:63]: Revert this, this script is not used

File: scripts/dialogue_system/logic/dialogue_group_player_base.gd [hunk 5]
```diff
@@ -18,7 +18,13 @@ func play_sequence_from_group(group: DialogueGroup) -> bool:
 	
 	_current_group = group
 	if not is_busy():
+		print("[" + str(Time.get_unix_time_from_system() * 1000) + "] Delay: " + str(group.initial_delay) + "s")
```
Comment (Marc313) [inline scripts/dialogue_system/logic/dialogue_group_player_base.gd:21]: Remove prints, kan ook later

File: scripts/dialogue_system/logic/dialogue_sequence_player.gd [hunk 6]
```diff
@@ -15,10 +17,13 @@ signal on_interruption_ready
 @onready var _animation_player =  $"../../AnimationPlayer" as OpponentAnimationPlayer
 @onready var _subtitle_displayer = $"../../Subtitle_System" as DialogueSubtitles
```
Comment (Marc313) [inline scripts/dialogue_system/logic/dialogue_sequence_player.gd:19]: Remove max_reps

File: scripts/material_highlighter.gd [hunk 7]
```diff
@@ -1,4 +1,4 @@
-@tool
```
Comment (Marc313) [inline scripts/material_highlighter.gd:1]: Revert

File: scripts/opponent_npc.gd [hunk 8]
```diff
@@ -95,3 +107,7 @@ func _on_ai_move_picker__on_suboptimal_move():
 
 func _on_move_picker__on_opponent_piece_captured():
 	pass # Replace with function body.
+
```
Comment (Marc313) [inline scripts/opponent_npc.gd:110]: Remove

**Review by Marc313 — CHANGES_REQUESTED** [review]

File: scenes/game/singleplayer.tscn (hunk 3 above)
Comment (dansebmey) [reply scenes/game/singleplayer.tscn:194]: ye

**Pull request merged**